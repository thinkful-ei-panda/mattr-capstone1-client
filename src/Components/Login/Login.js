import React, { Component } from 'react';
import '../RegistrationForm/RegistrationForm.css';
import ValidationError from '../ValidationError';
const validator = require("email-validator");


export default class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
        email: {
            value: '',
            touched: false
        },
        password: {
            value: '',
            touched: false
        }
      }
    }


    updateEmail(email) {
        this.setState({email: {value: email, touched: true}})
    };


    updatePassword(password) {
        this.setState({password: {value: password, touched: true}})
    };

    handleLogin(event){
        event.preventDefault();
        const { email, password } = this.state;

        console.log('email: ', email.value);
        console.log('Password: ', password.value);
    }


      validateEmail(){
          const email = this.state.email.value.trim();
          if(!validator.validate(email)) {
              return 'Must enter a valid email';
          }
      }

      validatePassword() {
        const password = this.state.password.value.trim();
        if (password.length === 0) {
          return 'Password is required';
        } else if (password.length < 6 || password.length > 72) {
          return 'Password must be between 6 and 72 characters long';
        } else if (!password.match(/[0-9]/)) {
          return 'Password must contain at least one number';
        }
      }


    render() {
    const emailError = this.validateEmail();
    const passwordError = this.validatePassword();

    return (
        <div>
            <form className="registration" action="/auth/login" onSubmit={e => this.handleLogin(e)}>

                <h2>Login</h2>
                
                <div className="form-group">
                <label htmlFor="email">E-mail *</label>
                <input type="email" className="registration__control" name="email" id="email" onChange={e => this.updateEmail(e.target.value)} />
                {this.state.email.touched && <ValidationError message={emailError} />}
                </div>
                

        <div className="form-group">
            <label htmlFor="password">Password *</label>
            <input type="password" className="registration__control" name="pw" id="password" onChange = {e => this.updatePassword(e.target.value)} />
            {this.state.password.touched && <ValidationError message={passwordError} />}
            <div className="registration__hint">6 to 72 characters, must include a number</div>
        </div>

        <div className="registration__button__group">
          <button type="reset" className="registration__button"> Reset </button>
          <button type="submit" className="registration__button" disabled={ this.validateEmail() || this.validatePassword() } > Login </button>
        </div>
      </form>

    </div>
    )}
}
