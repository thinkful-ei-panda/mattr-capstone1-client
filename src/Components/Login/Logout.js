import React, { Component } from 'react'
import TokenService from '../../Services/token-service'
import history from '../../history'
import Nav from '../Nav'
import '../App/App.css'
export default class Logout extends Component {

    handleLogout(event){
        event.preventDefault()
        TokenService.clearAuthToken()
        return history.push('/Home')
    }
    
    render() {

        return (
            <form className='VoteForm'>
                 <Nav />  
                <button className="vote-link" onClick={this.handleLogout} >
                    Logout
                </button>
                
            </form>
        )
    }
}
