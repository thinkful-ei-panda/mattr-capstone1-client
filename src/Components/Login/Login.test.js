import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Login from './Login';

describe(`Login Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <Login />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <Login />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})