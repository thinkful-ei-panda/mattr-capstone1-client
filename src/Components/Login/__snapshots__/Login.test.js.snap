// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="login-box"
>
  <div
    className="Nav"
  >
    <a
      href="/Home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/Election"
      onClick={[Function]}
    >
      Elections
    </a>
    <a
      href="/RegistrationForm"
      onClick={[Function]}
    >
       
      Register
       
    </a>
    <a
      href="/Login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
  <h2>
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="user-box"
    >
      <label>
        Email *
      </label>
      <input
        id="user_email"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
    <div
      className="user-box"
    >
      <label>
        Password *
      </label>
      <input
        id="user_password"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
    <button
      className="vote-link"
      type="submit"
    >
      <span />
      <span />
      <span />
      <span />
      Submit
    </button>
  </form>
</div>
`;
